owl/EASE-IO.owl

```diff
@@ -33,16 +33,27 @@
         <owl:equivalentClass>
             <owl:Class>
                 <owl:intersectionOf rdf:parseType="Collection">
-                    <rdf:Description rdf:about="http://www.ease-crc.org/ont/EASE-IO.owl#DigitalObject"/>
-                    <rdf:Description rdf:about="http://www.ontologydesignpatterns.org/ont/dul/DUL.owl#Agent"/>
+                    <rdf:Description rdf:about="http://www.ease-crc.org/ont/EASE-IO.owl#SoftwareImplementation"/>
+                    <rdf:Description rdf:about="http://www.ontologydesignpatterns.org/ont/dul/DUL.owl#PhysicalAgent"/>
                 </owl:intersectionOf>
             </owl:Class>
         </owl:equivalentClass>
+        <rdfs:comment>An agent that is implemented in software.</rdfs:comment>
```

Comment:
I think I like the original approach of having ComputationalAgent be an Agent, but not necessarily a Physical one.

Is Alexa a ComputationalAgent? Intuitively yes. Is it a Physical one? This is tricky, as I can see people classing it either way.

In any case, if a computational agent must be physical, then it cannot only be implemented in software. There needs to be a hardware implementation to support it.

Reply:
My argument was that the agent allocates memory on the computer device, and that the space region in the memory where the agent is represented is a physical manifestation of the agent. I do not insist on that, but I definitely need *InformationRealization* concepts with object character such that they can participate in events.
That's why I introduced *DigitalObject* which is a super-class of *ComputationalAgent*.
Digital object already is subclass of *PhysicalObject*, so in case we do not want to commit to computational agents being physical objects, we also need to touch this definition.

My motivation comes from IOLite where a distinction is made between *InformationObject* and *InformationRealization*.
IOLite comments that one example is the conceptualization of laws in some context (InformationObject) vs. a book with the laws written down (InformationRealization).
In the digital world we have *DataStructure* (InformationObject) vs. *DigitalResource* (InformationRealization).

Here, I would see *ComputationalAgent* analogous to *Robot* -- an entity that can solve some problems when appropriate input is specified, and when it is properly installed and running. In other words some piece of code that is currently executed on a computer, and that has an associated space region (however unknown) in the memory of the computer.
However, more distinctions could be possible like an installed but not running agent that also takes some space on the hard drive

Reply:
Ok.

Then a minor point on the rdfs:comment content: should it not be important that a computational agent be implemented in hardware as well? Or, if implemented is not the right relation, then bring hardware into the description somehow?

I ask because it seems to me that a casual use of implemented wouldn't necessarily imply your concept about how a computational agent takes up physical resources to exist and then be properly installed and running to act.

Reply:
I remove the physical stuff now, we can discuss this issue at a later point